ebos/eclthresholdpressure.hh

```diff
@@ -307,6 +337,45 @@ private:
                 }
             }
         }
+
+        if (enableExperiments) {
+            // apply threshold pressures accross faults (experimental!)
+            if (deck.hasKeyword("THPRESFT"))
+                extractThpresft_(deck.getKeyword("THPRESFT"));
+        }
+
+    }
+
+    void extractThpresft_(const Opm::DeckKeyword& thpresftKeyword)
+    {
+        // retrieve the faults collection.
+        const Opm::EclipseState& eclState = simulator_.vanguard().eclState();
+        const Opm::FaultCollection& faults = eclState.getFaults();
+
+        // extract the multipliers from the deck keyword
+        int numFaults = faults.size();
+        int numCartesianElem = eclState.getInputGrid().getCartesianSize();
+        thpresftValues_.resize(numFaults, -1.0);
+        cartElemFaultIdx_.resize(numCartesianElem, -1);
+        for (size_t recordIdx = 0; recordIdx < thpresftKeyword.size(); ++ recordIdx) {
+            const Opm::DeckRecord& record = thpresftKeyword.getRecord(recordIdx);
+
+            const std::string& faultName = record.getItem("FAULT_NAME").getTrimmedString(0);
```

Comment:
Whenever I get the chance I advocate that things like this should be in the `EclipseState` - and that the `Deck` should not be accessed outside of opm-common; the region based `THPRES` already has an implementation in the EclipseState object.

Reply:
okay, how to get to this information only using the `eclipseState`, then? I propose that this stays as-is until the parser properly internalizes THPRESFT. Also, be aware that this is clearly marked as an experimental feature.

Reply:
Well - you implement support in the EclipseState :-)

Reply:
Interesting theory ;)

Reply:
> Interesting theory ;)

Yes - a quite good plan; and frankly not a joke - although I see you take that way!

Reply:
> Whenever I get the chance I advocate that things like this should be in the EclipseState - and that the Deck should not be accessed outside of opm-common; the region based THPRES already has an implementation in the EclipseState object.

I guess I am guilty in violating the "not access the deck directly outside opm-common" rule several times. For simple keywords I love the way you can just add a JSON description and just use it in the simulator right away. For more complicated stuff when different keywords interact I think the "rule" should be obeyed strictly. 

I am open for merging this as is. But I also think we should have a discussion about the usage of the deck outside opm-common in general.